src/main/kotlin/com/team4099/robot2020/subsystems/LED.kt

```diff
@@ -0,0 +1,65 @@
+package com.team4099.robot2020.subsystems
+
+import com.team4099.lib.logging.HelixLogger
+import com.team4099.lib.subsystem.Subsystem
+import com.team4099.robot2020.config.Constants.LED.Color
+import com.team4099.robot2020.config.Constants
+import edu.wpi.first.wpilibj.AddressableLED
+import edu.wpi.first.wpilibj.AddressableLEDBuffer
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard
+
+object LED : Subsystem {
+
+    // PWM port 9
+    // Must be a PWM header, not MXP or DIO
+    var mLED = AddressableLED(Constants.LED.PWMPORT)
+    var currentColor = Color.ORANGE
+
+    // Reuse buffer
+    // Default to a length of 60, start empty output
+    // Length is expensive to set, so only set it once, then just update data
+    var mLEDBuffer = AddressableLEDBuffer(60)
+
+    private var initialRainbowHue: Int = 0
+
+    init {
+        mLED.setLength(mLEDBuffer.length)
+    }
+
+    private fun updateColor(wantedColor: Constants.LED.Color) {
+        for (i in 0 until mLEDBuffer.length) {
+            mLEDBuffer.setHSV(i, wantedColor.h, wantedColor.s, wantedColor.v)
+            if (wantedColor == Color.RAINBOW) {
+                wantedColor.h = (initialRainbowHue + (i * 180 / mLEDBuffer.length)) % 180
+                mLEDBuffer.setHSV(i, wantedColor.h, wantedColor.s, wantedColor.v)
+                initialRainbowHue += 6
+                initialRainbowHue %= 180
+            }
+        }
+        mLED.setData(mLEDBuffer)
```

Comment:
probably should add helix event saying color got changed

Reply:
wait plz don't do this: you're logging this every loop